### PR TITLE
Product service CQRS and functionality

### LINKED ISSUE #29: Add retrieve product functionality
**Description**
With this service a product's information can be retrieved.

**Why is this needed**
A description of why this is needed.

**Tasks**
- [ ] Implement main functionality
- [ ] Implement database functionality
- [ ] Implement RabbitMQ functionality

### PR DESCRIPTION
**Pull request by Rubenvdbrink**
## Changes
See checklist

## Checklist
- [x] configured rabbitmq correctly with the last up-to-date way (not sure if we're still gonna change the naming scheme but we will discuss that later)
- [x] Add add product, delete product commands
- [x] Add get product by id query

## When merging, this pull request 
closes #29 #45 #69 (nice) #44 #70

### CODE REVIEW

File: product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/driver/web/ProductDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import java.util.UUID;
+
+public record ProductDto(UUID id,
+                         double price,
+                         String name,
+                         String description,
+                         int weight,
+                         NutritionalValueDto nutritionalValueDto) {
+}
```
Comment (xandervedder) [inline product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/driver/web/ProductDto.java:11]: nit: inconsistent formatting with the other records (e.g. `CreateProductEvent`)

File: product/src/main/java/nl/softwarestrijders/waiter/product/core/port/data/ProductRepository.java [hunk 2]
```diff
@@ -6,4 +6,6 @@
 import java.util.UUID;
 
 public interface ProductRepository extends MongoRepository<Product, UUID> {
-}
+    boolean existsByPriceAndNameAndDescription(double price, String name, String description);
+    void deleteById(UUID id);
+}
```
Comment (xandervedder) [inline product/src/main/java/nl/softwarestrijders/waiter/product/core/port/data/ProductRepository.java:11]: nit: newline

**Review by xandervedder — DISMISSED** [review]

File: product/src/main/java/nl/softwarestrijders/waiter/product/core/application/QueryHandler.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    private final ProductRepository repository;
+
+    public QueryHandler(ProductRepository repository) {
+        this.repository = repository;
+    }
+
+    public Product handle(FindProductByIdQuery query) {
+        var id = query.id();
+        return this.repository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(id));
```
Comment (JustMilan) [inline product/src/main/java/nl/softwarestrijders/waiter/product/core/application/QueryHandler.java:20]: nit: the `var id = query.id();` is inconsistent with other methods. In other methods you do this inline.
See CommandHandler line 41 for example

If you don't see a problem here, just resolve it.

File: product/src/main/java/nl/softwarestrijders/waiter/product/core/port/data/ProductRepository.java (hunk 2 above)
Comment (JustMilan) [reply product/src/main/java/nl/softwarestrijders/waiter/product/core/port/data/ProductRepository.java:11]: yep

**Review by JustMilan — CHANGES_REQUESTED** [review]

File: product/src/main/java/nl/softwarestrijders/waiter/product/core/application/QueryHandler.java (hunk 3 above)
Comment (xandervedder) [reply product/src/main/java/nl/softwarestrijders/waiter/product/core/application/QueryHandler.java:20]: I don't see how that's relevant? `id` is reused multiple times here, hence it makes sense to put it in a variable. This is not the case for `CommandHandler::41`.

**Review by JustMilan — DISMISSED** [review]

File: product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/driver/web/ProductController.java [hunk 4]
```diff
@@ -1,27 +1,51 @@
...
+                        dto.fats(),
+                        dto.carbs(),
+                        dto.proteins(),
+                        dto.salts()
+                )
+        ));
+    }
+
+    @PostMapping("/delete{id}")
+    public void deleteProduct(@PathVariable String id) {
```
Comment (xandervedder) [inline product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/driver/web/ProductController.java:47]: `/delete/{id}`

**Review by xandervedder — APPROVED** [review]

**Review by arjennorbart — APPROVED** [review]

**Pull request merged**